ComplexTr/ComplexTr/ComplexTr.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+
+namespace ComplexTrr
+{
+    public struct ComplexTr
+    {
+        double abs;
+        public double Abs
+        {
+            get => abs;
+            set
+            {
+                if (value < 0)
+                    throw new ArgumentException("Значение должно быть неотрицательным");
+                abs = value;
+            }
+        }
+        public double Arg { get; set; }
+        public double Re
+        {
+            get => Math.Round(Abs * Math.Cos(Arg),13);
+        }
+        public double Im
+        {
+            get => Math.Round(Abs * Math.Sin(Arg),13);
+        }
+        public ComplexTr(double abs, double arg) : this()
+        {
+            Abs = abs;
+            Arg = arg;
+        }
+        public override string ToString()
+        {
+            if (Abs == 0)
+                return "0";
+            if (Abs == 1)
+                return $"z = cos ({Arg}) + i * sin ({Arg})";
+            else return $"z = {Abs} * (cos ({Arg}) + i * sin ({Arg}))";
+        }
+        public override bool Equals(object obj)
+        {
+            if (obj is ComplexTr)
+                return ((Abs == ((ComplexTr)obj).Abs)&&((Re+Im) - (((ComplexTr)obj).Re + ((ComplexTr)obj).Im))/(2*Math.PI) == 0);
```

Comment:
Неверный код. При сравнении модули и аргументы сравниваются отдельно. Кроме того, сравнение чисел с плавающей точкой нужно проводить с определенной точностью (указана в задании)
```suggestion
           if (obj is ComplexTr z)
                return (Math.Abs(Abs - z.Abs) < 1e-13 && 
                    Math.Abs((Im - z.Im) / (2 * Math.PI)) < 1e-13);
```